runtime/lua/vim/_defaults.lua

```diff
@@ -107,22 +107,50 @@ do
           vim.inspect(cmd.cmd)
         )
       end
+      return err
+    end
 
-      if err then
-        vim.notify(err, vim.log.levels.ERROR)
+    local function do_open_shotgun()
+      local current_node =
+        vim.treesitter.get_node { ignore_injections = false, lang = 'markdown_inline' }
+
+      while current_node do
+        local type = current_node:type()
+        if type == 'link_destination' then
+          return do_open(vim.treesitter.get_node_text(current_node, 0))
```

Comment:
if the function is renamed to `get_url`, then this becomes
```suggestion
          return vim.treesitter.get_node_text(current_node, 0)
```

and the caller calls `do_open(get_url())`.